confignode/src/main/java/org/apache/iotdb/confignode/manager/load/balancer/region/CopySetRegionAllocator.java

```diff
@@ -42,11 +42,9 @@ public class CopySetRegionAllocator implements IRegionAllocator {
 
   private int maxId = 0;
   private int intersectionSize = 0;
-  private final List<TDataNodeLocation> weightList;
+  private List<TDataNodeLocation> weightList;
 
-  public CopySetRegionAllocator() {
-    this.weightList = new ArrayList<>();
```

Comment:
why remove this to buildWeightList?

Reply:
In the same storage group, the same region type and different replicas, the region allocator is only created once. The weightList need to be initialized before building the weightList, otherwise it will be affected by the previous replica allocation.

Reply:
This is a recent change. In the past, every time a region is allocated, a new region allocator would be created.